source/distributed/tria.cc

```diff
@@ -1864,68 +1951,151 @@ namespace parallel
       const typename dealii::internal::p4est::types<dim>::forest
         *                parallel_forest,
       const char *       filename,
-      const unsigned int n_attached_deserialize)
+      const unsigned int n_attached_deserialize_fixed,
+      const unsigned int n_attached_deserialize_variable)
     {
+      // Large fractions of this function have been copied from
+      // DataOutInterface::write_vtu_in_parallel.
+      // TODO: Write general MPIIO interface.
+
       Assert(dest_data_fixed.size() == 0,
              ExcMessage("Previously loaded data has not been released yet!"));
 
-      const std::string fname_fixed = std::string(filename) + "_fixed.data";
+      variable_size_data_stored = (n_attached_deserialize_variable > 0);
 
-      // ----- copied -----
-      // from DataOutInterface::write_vtu_in_parallel
-      // TODO: write general MPIIO interface
       const int myrank = Utilities::MPI::this_mpi_process(mpi_communicator);
 
       MPI_Info info;
       int      ierr = MPI_Info_create(&info);
       AssertThrowMPI(ierr);
-      MPI_File fh;
-      ierr = MPI_File_open(mpi_communicator,
-                           const_cast<char *>(fname_fixed.c_str()),
-                           MPI_MODE_RDONLY,
-                           info,
-                           &fh);
-      AssertThrowMPI(ierr);
 
-      ierr = MPI_Info_free(&info);
-      AssertThrowMPI(ierr);
-      // ------------------
-
-      // Check if number of processors is lined up with p4est partitioning.
-      Assert(myrank < parallel_forest->mpisize, ExcInternalError());
-
-      // Read cumulative sizes from file.
-      // Since all processors need the same information about the data sizes,
-      // let each of them gain it by reading from the same location in the file.
-      sizes_fixed_cumulative.resize(1 + n_attached_deserialize);
-      ierr = MPI_File_read_at(fh,
-                              0,
-                              sizes_fixed_cumulative.data(),
-                              sizes_fixed_cumulative.size(),
-                              MPI_UNSIGNED,
-                              MPI_STATUS_IGNORE);
-      AssertThrowMPI(ierr);
+      //
+      // ---------- Fixed size data ----------
+      //
+      {
+        const std::string fname_fixed = std::string(filename) + "_fixed.data";
+
+        MPI_File fh;
+        ierr = MPI_File_open(
+          mpi_communicator, fname_fixed.c_str(), MPI_MODE_RDONLY, info, &fh);
+        AssertThrowMPI(ierr);
+
+        ierr = MPI_Info_free(&info);
+        AssertThrowMPI(ierr);
+
+        // Check if number of processors is lined up with p4est partitioning.
+        Assert(myrank < parallel_forest->mpisize, ExcInternalError());
```

Comment:
Make this an `AssertThrow` so that it also checks things in release mode -- you never know what kinds of input files people give to their codes...

Reply:
I don't think that this specific assertion is triggered by an input file.

`parallel_forest->mpisize` is just the number of mpi processes from the `mpi_communicator` registered with our `p::d::Triangulation`.

I would leave this one as is, but I agree on using `AssertThrow` on version checking.

Reply:
Oh yes, sorry. I meant to comment on the version check and just didn't pay attention.